src/promptflow-evals/promptflow/evals/synthetic/_model_tools/_async_http_client.py

```diff
@@ -0,0 +1,49 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+from aiohttp import TraceConfig
+from aiohttp_retry import RandomRetry, RetryClient
+
+
+class AsyncHTTPClientWithRetry:
+    def __init__(self, n_retry, retry_timeout, logger, retry_options=None):
+        self.attempts = n_retry
+        self.logger = logger
+
+        # Set up async HTTP client with retry
+
+        trace_config = TraceConfig()  # set up request logging
+        trace_config.on_request_end.append(self.delete_auth_header)
+        if retry_options is None:
+            retry_options = RandomRetry(  # set up retry configuration
+                statuses=[104, 408, 409, 424, 429, 500, 502, 503, 504],  # on which statuses to retry

```

Comment:
Should we retry on 424? I think AOAI returns it if it gets something unexpected, for example, gptv return it if the image is too big,

Reply:
I don't think we are supporting multi modal conversations, so its unlikely that we will run into 424. I'll remove it.